### ISSUE TITLE
Aarch64 to Risc-V64 translation

### ISSUE THREAD

**Issue by egdeeno** [aengelke/instrew#14]
Hello!
Could you, please, answer, is it possible to add Aarch64(guest) to Riscv64(host) translation in instrew?
If so, could you, please, describe steps and changes needed to implement such translation?

**Comment by aengelke** [comment]
RISC-V hosts are currently not supported, I don't have access to RISC-V hardware with reasonable performance.

Bringing RISC-V support up should be fairly easy.

- client/main.c: set new tsc_host_arch
- client/memory.c: implement ICache flushing in mem_write_code, this is a syscall on Linux (do not use fence.i, this is insufficient)
- client/rtld.c: implement PLT/relocations. Basically similar to AArch64, look for the ifdefs
- client/emulate.c: maybe something needs to be done for clone and other syscalls, I'm not sure.
- server/codegenerator.cc: set triple for RISC-V

For performance, you also want to look at calling conventions, where things typically get a bit messy.

**Comment by egdeeno** [comment]
Thank you!
Should client/minilibc.c be modified?

**Comment by aengelke** [comment]
Yeah, forgot about that. You need to implement `_start`, `__clone`, syscall wrappers, and dynamic relocations. Should not be substantially different from AArch64, though.

**Comment by egdeeno** [comment]
Thank you!

**Comment by yli147** [comment]
> Thank you!

do you have any progress on this ?
